application/src/main/java/org/opentripplanner/street/model/edge/StreetEdge.java

```diff
@@ -1009,14 +1009,16 @@ private StateEditor doTraverse(State s0, TraverseMode traverseMode, boolean walk
        * that during reverse traversal, we must also use the speed for the mode of
        * the backEdge, rather than of the current edge.
        */
+      var intersectionMode = arriveBy ? backMode : traverseMode;
```

Comment:
Do we have some test(s) that ensure that this forward/backward search will work?